### PR TITLE
Add support for very long period timers

### PR DESCRIPTION
**Pull request by anttiylitokola**
M2MTimer allows to pass uint64 values so adding internal handling of very long timer values.

### CODE REVIEW

**Comment by anttiylitokola** [comment]
@yogpan01, @TeroJaasko

File: source/m2mtimerpimpl.cpp [hunk 1]
```diff
@@ -73,19 +73,19 @@ M2MTimerPimpl::M2MTimerPimpl(M2MTimerObserver& observer)
...
   _total_interval(0),
+  _still_left(0),
   _status(0),
   _dtls_type(false)
 {
     ns_hal_init(NULL, MBED_CLIENT_EVENT_LOOP_SIZE, NULL, NULL);
     eventOS_scheduler_mutex_wait();
     if (_tasklet_id < 0) {
-        _tasklet_id = eventOS_event_handler_create(tasklet_func, MBED_CLIENT_TIMER_EVENT);
+        _tasklet_id = eventOS_event_handler_create(tasklet_func, 0 /*ARM_LIB_TASKLET_INIT_EVENT*/);
```
Comment (TeroJaasko) [inline source/m2mtimerpimpl.cpp]: Is the '0' left there by mistake?

Comment (anttiylitokola) [inline source/m2mtimerpimpl.cpp]: ARM_LIB_TASKLET_INIT_EVENT is defined in net_interface.h which lives under sal-stack-nanostack component. Adding that component will cause dependency to IPv6 stuff and that's not what we want?

Comment (TeroJaasko) [inline source/m2mtimerpimpl.cpp]: Ah, you are right. Then a "#define MBED_CLIENT_TIMER_TASKLET_INIT_EVENT 0" or similar somewhere around MBED_CLIENT_TIMER_EVENT definition might document the intention of the event id and avoid making them the same by mistake.

File: mbed-client-classic/m2mtimerpimpl.h [hunk 2]
```diff
@@ -80,8 +79,23 @@ class M2MTimerPimpl {
      */
     bool is_total_interval_passed();
 
+    /**
+     * @brief Start long period timer
+     */
+    void start_still_left_timer();
```
Comment (TeroJaasko) [inline mbed-client-classic/m2mtimerpimpl.h]: Should this method be private, or is there a place in client side where caller needs to call it explicitly?

File: source/m2mtimerpimpl.cpp (hunk 1 above)
Comment (anttiylitokola) [inline source/m2mtimerpimpl.cpp]: Yes, I'll create own define for the init event.

**Comment by anttiylitokola** [comment]
Compilation fails due to this error:
https://github.com/ARMmbed/mbed-os/issues/2653

**Comment by anttiylitokola** [comment]
rebuild

**Comment by anttiylitokola** [comment]
rebuild

**Comment by anttiylitokola** [comment]
rebuild

**Pull request merged**